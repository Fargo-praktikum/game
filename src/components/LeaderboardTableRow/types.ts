import { UserScore } from "../../models/userScore";

export interface LeaderboardTableRowProps {
    userData: UserScore;
}
