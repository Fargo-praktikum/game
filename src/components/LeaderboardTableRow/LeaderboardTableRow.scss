.tablerow {
    padding: 15px 5px;
    display: table-cell;
    text-align: left;
    vertical-align: middle;
    border-radius: 2px;

    &__img {
        height: 50px;
        width: 50px;
        border-radius: 50%;
        fill: #999;
    }

    &__link {
        display: flex;
        align-items: center;
        text-align: center;
    }

    &__name {
        padding-left: 10px;
    }

    &__scoreNumber {
        padding-left: 20px;
    }
}

.scoreNumber {
    width: 100px;
}

