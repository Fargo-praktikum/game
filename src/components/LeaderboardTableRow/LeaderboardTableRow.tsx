import React from "react";
import "./LeaderboardTableRow.scss";
import avatar from "../../assets/circle.png";
import { LeaderboardTableRowProps } from "./types";

export const LeaderboardTableRow = (props: LeaderboardTableRowProps): JSX.Element => {
    const { id, name, theme, score } = props.userData;

    return (    <tr className="table-wrapper__line">
        <td className="tablerow">{id}</td>
        <td className="tablerow" align="left">
            <a target="_blank" className="tablerow__link">
                <img className="tablerow__img" src={avatar}
                    width="50" height="50" alt="Image not found"/>
                <span className="tablerow__name">{name}</span>
            </a>
        </td>
        <td className="tablerow">{theme}</td>
        <td className="tablerow">{score}</td>
    </tr>);
};
