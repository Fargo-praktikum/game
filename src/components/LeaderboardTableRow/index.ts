import { LeaderboardTableRow } from "./LeaderboardTableRow";

export default LeaderboardTableRow;
