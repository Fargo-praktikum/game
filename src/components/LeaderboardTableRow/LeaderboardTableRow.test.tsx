import React from "react";

import renderer from "react-test-renderer";
import { LeaderboardTableRow } from "./LeaderboardTableRow";

describe("Проверка отрисовки строки лидерборда", () => {

    it("Отрисовка строки лидерборда с данными", () => {
        const userData = {
            rating: 1,
            name: "test",
            date: 1,
            score: 1,
            userId: 1,
            theme: "test",
        };

        const tree = renderer
            .create(<LeaderboardTableRow userData={userData}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});



