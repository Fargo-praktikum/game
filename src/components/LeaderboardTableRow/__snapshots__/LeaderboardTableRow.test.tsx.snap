// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка отрисовки строки лидерборда Отрисовка строки лидерборда с данными 1`] = `
<tr
  className="table-wrapper__line"
>
  <td
    className="tablerow"
  >
    <img
      alt="1 place"
      className="tablerow__img"
      src="test-file-stub"
    />
  </td>
  <td
    align="left"
    className="tablerow"
  >
    <a
      className="tablerow__link"
      target="_blank"
    >
      <img
        alt="Image not found"
        className="tablerow__img"
        height="50"
        src="test-file-stub"
        width="50"
      />
      <span
        className="tablerow__name"
      >
        test
      </span>
    </a>
  </td>
  <td
    className="tablerow"
  >
    1/1/1970
  </td>
  <td
    className="tablerow"
  >
    1
  </td>
</tr>
`;
